aptos-move/framework/aptos-names/sources/domains.move

```diff
@@ -148,13 +153,23 @@ module aptos_names::domains {
         assert!(is_valid, error::invalid_argument(EDOMAIN_HAS_INVALID_CHARACTERS));
         assert!(length <= config::max_domain_length(), error::out_of_range(EDOMAIN_TOO_LONG));
 
+        let account_address = signer::address_of(sign);
+        if (!unrestricted_mint_enabled()) {
+            assert!(vector::length(&signature) == 64, error::invalid_argument(EVALID_SIGNATURE_REQUIRED));
+            verify::verify_register_domain_signature(signature, account::get_sequence_number(account_address), account_address, domain_name);
+        };
 
         let price = price_model::price_for_domain_v1(length, num_years);
         coin::transfer<AptosCoin>(sign, config::foundation_fund_address(), price);
 
         register_name_internal(sign, subdomain_name, domain_name, registration_duration_secs, price);
         // Automatically set the name to point to the sender's address
-        set_name_address_internal(subdomain_name, domain_name, signer::address_of(sign));
+        set_name_address_internal(subdomain_name, domain_name, account_address);
+    }
+
+    /// Enforcing signature for `register_domain`
+    public entry fun register_domain_with_signature(sign: &signer, domain_name: String, num_years: u8, register_domain_signature: vector<u8>) acquires NameRegistryV1, RegisterNameEventsV1, SetNameAddressEventsV1 {
```

Comment:
what's the purpose of this method now?

Reply:
oops just updated! now it's forcing a signature parameter of length 64

Reply:
this method is forcing them to have a 64-byte signature, otherwise would abort. 

in `register_domain`, we check `if (!unrestricted_mint_enabeld() || vector::length(&signature) == 64)` - if we call `register_domain` with a `signature` of incorrect length (could be empty) when `unrestricted_mint_enabled() == true` then they can skip the signature verification. If we call `register_domain_with_signature` with a `signature` with incorrect length when `unrestricted_mint_enabled() == true`, then it would abort.  

Yes it's a bit redundant, but by switching the functions on the backend we can enforce the signature (although if we turn `unrestricted_mint_enabled()` to false we can do the same, but I guess it's one more way to do this?

Reply:
I find it confusing and possibly error-prone.

I propose the following cleaner design:
1. `register_domain` takes no signature and fails if `unrestricted_mint_enabled() == false`.
2. `register_domain_with_signature` takes the signature and always verifies it before registering, even when `unrestricted_mint_enabled() == true`
3. You might need a private, internal helper function that both of these functions call which does the actual registration.

Please do not keep the current confusing design where `register_domain_with_signature` calls into `register_domain` though. It's very hard to tell what's going on.

Reply:
okay, I see why it's confusing, updating soon

Reply:
for the signature length check, i was thinking that the length will be 64 cus the public key will be an ed25519 key, but i can drop it and keep it generic.

Reply:
But we should stay flexible for the future. Plus, the signature verification API will validate everything anyway, including length. So adding that length check here is redundant and potentially conflicts with future changes, even though it (might) serve to reduce gas costs by aborting early on an incorrectly-sized signature (but that would be best implemented in the signature verification API).

Reply:
okay just updated : )